azurerm/internal/services/storage/resource_arm_hpc_cache_blob_target.go

```diff
@@ -0,0 +1,272 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	storage "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/client"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/parsers"
+	storageValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+	"github.com/tombuildsstuff/giovanni/storage/2018-11-09/blob/containers"
+)
+
+func resourceArmHPCCacheBlobTarget() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHPCCacheBlobTargetCreateOrUpdate,
+		Update: resourceArmHPCCacheBlobTargetCreateOrUpdate,
+		Read:   resourceArmHPCCacheBlobTargetRead,
+		Delete: resourceArmHPCCacheBlobTargetDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parsers.HPCCacheTargetID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: storageValidate.HPCCacheTargetName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"cache_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"namespace_path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: storageValidate.HPCCacheNamespacePath,
+			},
+
+			"storage_container_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func resourceArmHPCCacheBlobTargetCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Storage.StorageTargetsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure HPC Cache Blob Target creation.")
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	cache := d.Get("cache_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
```

Comment:
we can remove this since this feature's always on now:

```suggestion
	if d.IsNewResource() {
```